Objects/codeobject.c

```diff
@@ -1884,9 +1887,21 @@ code_hash(PyCodeObject *co)
     SCRAMBLE_IN(Py_SIZE(co));
     for (int i = 0; i < Py_SIZE(co); i++) {
         int deop = _Py_GetBaseOpcode(co, i);
-        SCRAMBLE_IN(deop);
-        SCRAMBLE_IN(_PyCode_CODE(co)[i].op.arg);
-        i += _PyOpcode_Caches[deop];
+        if (deop == ENTER_EXECUTOR) {
+            // Assume that deop of ENTER_EXECUTOR will be ENTER_EXECUTOR.
+            const int exec_index = _PyCode_CODE(co)[i].op.arg;
+            _PyExecutorObject *exec = co->co_executors->executors[exec_index];
+            assert(exec != NULL);
+            assert(exec->vm_data.opcode != ENTER_EXECUTOR);
+            SCRAMBLE_IN(exec->vm_data.opcode);
+            SCRAMBLE_IN(exec->vm_data.oparg);
+            i += _PyOpcode_Caches[exec->vm_data.opcode];
+        }
+        else {
+            SCRAMBLE_IN(deop);
+            SCRAMBLE_IN(_PyCode_CODE(co)[i].op.arg);
+            i += _PyOpcode_Caches[deop];
+        }
```

Comment:
It would be nicer if this was written like the compare code, where we first extract the opcode+oparg, then fix both if opcode==ENTER_EXECUTOR, and finally do what we want with the fixed opcode+oparg.

Reply:
Done!